### PR TITLE
kotlin Migration of cloze Test

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Kotlin migration of clozeTest .

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/ClozeTest.kt [hunk 1]
```diff
@@ -1,115 +1,125 @@
...
+        } catch (e: JSONException) {
+            fail()
         }
         // a cloze model with no clozes is not empty
-        f.setItem("Text", "nothing");
-        assertThat(d.addNote(f), is(greaterThan(0)));
-        Card card = f.cards().get(0);
-        assertTrue(card.isEmpty());
+        f.setItem("Text", "nothing")
+        assertThat(d.addNote(f), Matchers.`is`(Matchers.greaterThan(0)))
```
Comment (ShridharGoel) [inline AnkiDroid/src/test/java/com/ichi2/libanki/ClozeTest.kt:29]: Import the method directly

File: AnkiDroid/src/test/java/com/ichi2/libanki/ClozeTest.kt [hunk 2]
```diff
@@ -1,115 +1,124 @@
...
+        assertTrue(card.isEmpty)
         // try with one cloze
-        f = d.newNote(d.getModels().byName("Cloze"));
-        f.setItem("Text", "hello {{c1::world}}");
-        assertEquals(1, d.addNote(f));
-        assertThat(f.firstCard().q(), containsString("hello <span class=cloze>[...]</span>"));
-        assertThat(f.firstCard().a(), containsString("hello <span class=cloze>world</span>"));
+        f = d.newNote(d.models.byName("Cloze"))
+        f.setItem("Text", "hello {{c1::world}}")
+        assertEquals(1, d.addNote(f).toLong())
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/ClozeTest.kt:34]: `toLong()` gets added incorrectly by the conversion, could you remove it

File: AnkiDroid/src/test/java/com/ichi2/libanki/ClozeTest.kt [hunk 3]
```diff
@@ -1,115 +1,124 @@
...
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.ichi2.anki.RobolectricTest
+import com.ichi2.utils.JSONException
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.*
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
 
+@RunWith(AndroidJUnit4::class)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/ClozeTest.kt:14]: `@KotlinCleanup("is -> equalTo")`

**Review by david-allison — CHANGES_REQUESTED** [review]
nits, then good to go

**Review by mikehardy — APPROVED** [review]
Looks good thanks!

**Review by david-allison — APPROVED** [review]

**Pull request merged**